### PR TITLE
Exporting all from Sketch

### PR DESCRIPTION
**Pull request by kardeslik**
New export abilities:
- All artboards from the current page
- All artboards from all pages
- All symbols from the current page
- All symbols from all pages

Screen Shot 2019-07-11 at 18 33 23

Edit: Solves https://github.com/zeplin/zeplin-sketch-plugin/issues/16

### CODE REVIEW

File: Zeplin.sketchplugin/Contents/Sketch/export.cocoascript [hunk 1]
```diff
@@ -42,282 +44,85 @@ var onRun = function (context) {
...
-    artboardNamesByIdentifier = nil;
-    
-    writeDirectives(directives, path);
-
-    directives = nil;
-
-    launchZeplin(context, path);
-
-    path = nil;
+    exportArtboards(context, artboards);
```
Comment (berkcebi) [inline Zeplin.sketchplugin/Contents/Sketch/export.cocoascript]: We should check for `foreignSymbolsUpToDate` on all artboard/symbol exports, so moving that to `exportArtboards` would make sense.

File: Zeplin.sketchplugin/Contents/Sketch/manifest.json [hunk 2]
```diff
@@ -48,16 +76,28 @@
         "name": "Shortcut Help",
         "identifier": "shortcut-help",
         "shortcut": "cmd e",
-        "script": "export.cocoascript",
+        "script": "utils.cocoascript",
```
Comment (berkcebi) [inline Zeplin.sketchplugin/Contents/Sketch/manifest.json]: Here's an idea: We can move the methods in `utils` to `export` and rename `export` to something like`main` or `index`. In that case, we can rename `helpers` to `utils`, which would make more sense. 😅

File: Zeplin.sketchplugin/Contents/Sketch/utils.cocoascript [hunk 3]
```diff
@@ -37,3 +37,13 @@ var includeSublayers = function (context) {
...
+
+var shortcutHelp = function (context) {
+    var alert = [NSAlert alertWithMessageText:@"Shortcut changed to ⌃⌘E" defaultButton:@"Continue and Export" alternateButton:@"Cancel" otherButton:nil informativeTextWithFormat:@"Due to recent shortcuts tweaks on Sketch, Zeplin's “Export Selected Artboards…” shortcut is now ⌃⌘E.\n\n☝️ That's Command, Control, E."];
+    
+    if ([alert runModal] == NSAlertDefaultReturn) {
+        onRun(context);
+    }
+    
+    alert = nil;
+}
```
Comment (berkcebi) [inline Zeplin.sketchplugin/Contents/Sketch/utils.cocoascript]: We can completely remove shortcut help actually!

File: script/install.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        rm -rf "$SKETCH_PLUGIN_PATH"
+        cp -r "$TEMP_PLUGIN_PATH" "$SKETCH_PLUGIN_PATH"
+
+        open "$SKETCH_APP_PATH"
+    else
+        open "$TEMP_PLUGIN_PATH"
+    fi
+
+    exit 0
+}
```
Comment (berkcebi) [inline script/install.sh]: Mind summarizing what the script is doing? Just to make sure that I understand it fully.

Comment (kardeslik) [reply script/install.sh]: You know we have different plugins for different Zeplin app configurations. They have different names, identifiers and Zeplin app bundle identifiers.
This script makes necessary changes on aforementioned values according to the given type and install it to the sketch.
Install is done in two ways according to the availability of the plugin:
- If a plugin with same name is already installed: Copy the plugin to the sketch's plugin directory and open Sketch.
- If this is the first install: Just open the plugin file and let Sketch app take the rest.

I hope my explanation is clear enough, if not, let's discuss it offline. 😅

Comment (berkcebi) [reply script/install.sh]: Got it. This makes sense but I just realized that Sketch users that end up here at GitHub (through the support articles and such), they download the repo, so they would see the script file.

For now, let's use this script internally (remove it from the repo) and we can figure out a better solution later!

File: Zeplin.sketchplugin/Contents/Sketch/manifest.json [hunk 5]
```diff
@@ -11,60 +11,90 @@
...
         "name": "Export Text Styles…",
         "identifier": "exportTextStyles",
-        "script": "export.cocoascript",
+        "script": "main.cocoascript",
         "handler": "exportTextStyles",
         "icon": "Icons/icZeplinRunner.png",
         "description": "Export text styles to Zeplin."
+    }, {
+        "name": "Artboards from Current Page…",
+        "identifier": "artboardsFromCurrentPage",
```
Comment (yigitcanyurtsever) [inline Zeplin.sketchplugin/Contents/Sketch/manifest.json]: Shouldn't we have the identifiers for the export items start with `export` prefix as well? i.e. `"exportArtboardsFromCurrentPage"`

Comment (kardeslik) [reply Zeplin.sketchplugin/Contents/Sketch/manifest.json]: These ones are under the `Export All` menu therefore we didn't put `Export` word to he start of the names and thus I didn't put it to the identifier as prefix.

**Pull request merged**